subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -4762,8 +4762,13 @@ int lwm2m_parse_peerinfo(char *url, struct sockaddr *addr, bool *use_dtls)
 	}
 
 	if (!(parser.field_set & (1 << UF_PORT))) {
-		/* Set to default port of CoAP */
-		parser.port = CONFIG_LWM2M_PEER_PORT;
+		if (is_firmware_uri) {
```

Comment:
If _UF_PORT_ is not available we should use _UF_SCHEMA_ to decide between ports 5683(coap) and 5684(coaps). In other words _use_dtls_ variable should be checked here.

Reply:
What do you think @rlubos? selecting 5683 or 5684 based on the schema means not having a configurable firmware port, right?

Reply:
Hmm, I'm not convinced - this would basically mean that we hardcode the port number to the default one. 
While on one hand it may sound reasonable to fallback to the default port in case no port is present in the URI, on the other I think it's very likely that he FW would be hosted on a different port, for example if the CoAP server hosting FW runs on the same host. I think it's highly server-dependent, therefore I'd prefer to keep the default port configurable. We might consider specifying separate port config for secure/nonsecure transfers, but IMO it should rather be expected from the LwM2M server to provide the port number, and use the configurable default value really as a last resort.

Reply:
Yeah, I agree

Reply:
> We might consider specifying separate port config for secure/nonsecure transfers

Sounds good. Let's make it so.

Reply:
Would that be for just the firmware port, or the peer port as well? So in total 4 config options for default ports?

Reply:
My opinion is that peer port could stay as it is.